status/projects.py

```diff
@@ -125,20 +184,21 @@ def project_summary_data(self, row):
 
         return row
 
-    def list_projects(self, filter_projects='all', oldest_date='2012-01-01', youngest_date=datetime.datetime.now().strftime("%Y-%m-%d")):
+    def list_projects(self, filter_projects='all', start_date=(datetime.datetime.now() - relativedelta(years=2)), end_date=datetime.datetime.now().strftime("%Y-%m-%d")):
```

Comment:
As we discussed when you re-write this function, you could also use the date efficiently by sorting it first before iterating on all the project. Cause then if the "start_open_date" is only 2 years till today, then while iterating on sorted list of projects (descending), then you can break the iteration when a projects open_date is less than the "start_open_date".

For example, in order to get all open projects within 2 years currently it will iterate through all 3220 projects (at the moment). But when you do sort and break, you would have only iterated over 1192 projects (projects open for past two years from today)

**Note:** This is just a draft suggestion to give you an idea, of course this doesn't cover different scenarios. But when you refactor this function you can implement it in such way that it handles all 👍